include/cucumber-cpp/internal/step/StepMacros.hpp

```diff
@@ -25,13 +26,45 @@ ::cucumber::internal::registerStep< step_name >(         \
 )                                                        \
 /**/
 
+#if defined(CUKE_ENABLE_BREAKPOINT_SUPPORT)
+#define BODY_WITH_BREAKPOINT_IMPLEMENTATION_(step_class_name)                 \
+  {                                                                           \
+    if (BreakpointSupport::checkAndClearShouldBreakIntoDebugger()) {          \
+      CUKE_BREAK_INTO_DEBUGGER;                                               \
+    }                                                                         \
+    bodyIfCompiledWithBreakpointSupport();                                    \
+  }                                                                           \
+  void step_class_name ::bodyIfCompiledWithBreakpointSupport()
+/**/
+
+#define CUKE_STEP_WITH_BREAKPOINT_WITH_NAME_(step_name, step_matcher)         \
+  CUKE_STEP_WITH_NAME_(step_name, step_matcher)                               \
+  BODY_WITH_BREAKPOINT_IMPLEMENTATION_(step_name)
+/**/
+
+#define CUKE_STEP_WITH_BREAKPOINT_(step_matcher)                              \
```

Comment:
If we really need conditional breakpoint support (I don't think so), wouldn't it be nicer to just give two definitions of `CUKE_STEP_` instead of having this separate `CUKE_STEP_WITH_BREAKPOINT_ `?